chain/chainhandle.go

```diff
@@ -1039,7 +1052,7 @@ func executeTx(execCtx context.Context, ccc consensus.ChainConsensusCluster, cdb
 			}
 			return types.ErrNotAllowedFeeDelegation
 		}
-		rv, events, txFee, err = contract.Execute(execCtx, bs, cdb, tx.GetTx(), sender, receiver, bi, executionMode, true)
+		rv, events, _, txFee, err = contract.Execute(execCtx, bs, cdb, tx.GetTx(), sender, receiver, bi, executionMode, true)
```

Comment:
feeDelegation executes the contract like a normal transaction execution, except that the contract wallet pays the transaction fee. In this case, is it not necessary to record internalOps? Or is there no possibility of internalOps occurring in a feeDelegation transaction?

Reply:
Good catch!

Transactions with fee delegation are executed in 2 steps:

1. (the code above) the mempool executes the contract (`check_delegation` function) to check if it will pay for the fees
2. if yes, then the txn is included in a block and then executed normally

Currently the code is logging the internal operations only from the normal execution

I also suspect that on step 1 it is not allowed to make some operations like call other contracts, deploy, etc. It can only read from the state variables.